src/main/java/Duke.java

```diff
@@ -21,24 +21,94 @@ public static void main(String[] args) {
                 String byeText = "Bye. Hope to see you again soon!";
-                System.out.println(textLine + byeText + textLine);
+                System.out.println(byeText);
                 break;
             }
-            System.out.println(textLine + input + textLine);
+            else if (input.equals("list")){
+
+                for (int index = 1; index < counter; index += 1){
+                    System.out.println((index) + "." + "[" + tasks[index].getStatusIcon() + "]" + " " + tasks[index].getDescription());
+                }
+            }
+            else if (input.substring(0, 4).equals("mark")) {
+                if (input.length() <= 5) {
+                    System.out.println("Please enter a valid number after space, e.g. mark 5");
+                    continue;
+                }
+                String indexString = input.substring(5);
+                indexString = indexString.replaceAll("\\s", "");
+                int index = Integer.parseInt(indexString);
+                if (index < counter && index != 0) {
+                    if (tasks[index].isDone())  {
+                        System.out.println("  It is already done silly! Do something else...");
+                        continue;
+                    }
+                    else {
+                        tasks[index].markAsDone();
+                        System.out.println("  Nice! I've marked this task as done:\n"+"    [" +
+                                tasks[index].getStatusIcon() + "]" + " " + tasks[index].getDescription());
+                    }
+
+                }
+                else {
+                    System.out.println("Please enter a valid number");
+                    continue;
+                }
+
+            }
+            else if (input.substring(0, 6).equals("unmark")) {
+                if (input.length() <= 5) {
+                    System.out.println("Please enter a valid number after space, e.g. mark 5");
+                    continue;
+                }
+                String indexString = input.substring(7);
+                indexString = indexString.replaceAll("\\s", "");
+                int index = Integer.parseInt(indexString);
+                if (index < counter && index != 0) {
+                    if (!tasks[index].isDone())  {
+                        System.out.println("  It is not done silly! Do it quickly!");
+                        continue;
+                    }
+                    else {
+                        tasks[index].markAsNotDone();
+                        System.out.println("  Okay... whatever you say sir.... unmarked!\n"+"    [" +
+                                tasks[index].getStatusIcon() + "]" + " " + tasks[index].getDescription());
+                    }
+
+                }
+                else {
+                    System.out.println("Please enter a valid number");
+                    continue;
+                }
+            }
+            else {
```

Comment:
Maybe you can try to move your else one line up. For example, else in line 97 move up to line 96.